File: golden_ticket_enterprise/lib/entities/rating.dart

import 'package:golden_ticket_enterprise/entities/chatroom.dart';

class Rating {
  final int ratingID;
  Chatroom chatroom;
  int score;
  String? feedback;
  DateTime createdAt;
  Rating({ required this.ratingID, required this.chatroom, required this.score, required this.feedback, required this.createdAt});

  factory Rating.fromJson(Map<String, dynamic> json) {
    return Rating(
      ratingID: json['ratingID'],
      chatroom: Chatroom.fromJson(json['chatroom']),
      score: json['score'],
      feedback: json['feedback'] ?? 'None Provided',
      createdAt: DateTime.parse(json['createdAt'])
    );
  }

}

File: GoldenTicket/GoldenTicket/Entities/Rating.cs
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using GoldenTicket.Entities;

namespace GoldenTicket.Entities
{
    [Table("tblRating")]
    public class Rating
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int RatingID { get; set; }
        [Required]
        public int ChatroomID { get; set; }
        [ForeignKey("ChatroomID")]
        public Chatroom? Chatroom { get; set; } = null;
        [Required]
        public DateTime CreatedAt { get; set;}
        [Required]
        public int Score { get; set; }
        public string? Feedback { get; set; } = "None provided";
    }

    public class RatingDTO {
        public int RatingID { get; set; }
        public ChatroomDTO Chatroom { get; set; }
        public int Score { get; set; }
        public string? Feedback { get; set; } = "None Provided";
        public DateTime CreatedAt { get; set; } 

        public RatingDTO(Rating rating){
            this.RatingID = rating.RatingID;
            this.Chatroom = new ChatroomDTO(rating.Chatroom!);
            this.Score = rating.Score;
            this.Feedback = rating.Feedback;
            this.CreatedAt = rating.CreatedAt;
        }
    }
}
